src/search/pdbs/pattern_collection_generator_hillclimbing.cc

```diff
@@ -264,8 +264,9 @@ pair<int, int> PatternCollectionGeneratorHillclimbing::find_best_improving_pdb(
 bool PatternCollectionGeneratorHillclimbing::is_heuristic_improved(
     const PatternDatabase &pdb, const State &sample, int h_collection,
     const PDBCollection &pdbs, const vector<PatternClique> &pattern_cliques) {
+    const vector<int> &sample_data = sample.get_unpacked_values();
```

Comment:
I think it would be better to have the `unpack` and `get_unpacked_values` method calls close together. In this code snippet, for example, it's not obvious whether the call to `get_unpacked_values` will lead to an error or not. I see three alternatives:
* Call `unpack` explicitly in this method.
* Pass `const vector<int> &` to the method.
* Combine `unpack` and `get_unpacked_values` into one method that unpacks the data if it's not unpacked and then returns the const reference.

I'm sure you already discussed the third option, so I'd go for the second option (as is done in the file below).

Reply:
If we pass in a `vector<int>` here the same comment would apply to the method calling this function. At some point we have to make the step from states to vectors and I think it is better to do this as late as possible without impacting performance. Since this is a private method, I think it is fine to rely on the fact that samples used within the class are all unpacked states.